### ISSUE TITLE
$project step omits referenced properties when matching property exists in incoming data

### ISSUE THREAD

**Issue by cezarcarvalhaes** [kofrasa/mingo#406]
When running a pipeline operation where the `$project` step includes a property that already matches a property within the incoming data, referenced properties are omitted from the result. See the code sample below: 

```javascript
const operation = [
  {
    "$project": {
      "bar": {
        "id": "$foo.one",
        "static": "text"
      }
    }
  }
];

const agg = new Aggregator(operation)

const data = [
  {
    "foo": {
      "one": 1,
      "two": 2
    },
    "bar": {
      "one": 1,
      "two": 2
    }
  }
];

const result = agg.run(data);

console.log(result)
// expected: { bar: id: 1, static: 'text' }
// received: { bar: static: 'text' }

```

The above pipeline operation works as expected in mongo however: https://mongoplayground.net/p/GcomeC55FFb

If the projected property does not match an incoming data prop, everything works again:

```javascript
const operation = [
  {
    "$project": {
      "baz": { // <-- changed "bar" to "baz"
        "id": "$foo.one",
        "static": "text"
      }
    }
  }
];

const agg = new Aggregator(operation)

const data = [
  {
    "foo": {
      "one": 1,
      "two": 2
    },
    "bar": {
      "one": 1,
      "two": 2
    }
  }
];

const result = agg.run(data);

console.log(result)
// expected: { baz: id: 1, static: 'text' }
// received: { baz: id: 1, static: 'text' }

```

**Comment by kofrasa** [comment]
Thanks for reporting this issue. I just tested this in the current version and it works as expected. What version of the library are you using?

**Comment by kofrasa** [comment]
I think the bug has been fixed since [6.2.7](https://github.com/kofrasa/mingo/blob/main/CHANGELOG.md#627--2023-01-12)

**Comment by cezarcarvalhaes** [comment]
> Thanks for reporting this issue. I just tested this in version `>= 6.4.0` and it works as expected. What version of the library are you using?

My apologies, I was 6.1.2. Before reporting the bug, I updated to 6.4.9, but forgot to re-install my node_modules. Just re-ran tests and all is well. Thanks again!
